Pass context to FormatChecker

tung2744 opened the pull request:
ref DEV-1139
related pr: https://github.com/iawaknahc/jsonschema/pull/2
based on https://github.com/authgear/authgear-server/pull/5012

louischan-oursky commented:
Please use the latest commit of jsonschema, and rebase.

louischan-oursky commented on pkg/lib/images/model.go:
```diff
@@ -76,6 +77,7 @@ func DecodeFileMetadata(encoded string) (*FileMetadata, error) {
 	}
 
 	err = FileMetaSchema.Validator().ValidateWithMessage(
+		context.Background(),
```
Take ctx in this function.

louischan-oursky commented on cmd/portal/cmd/cmdinternal/migrate_resources_migrate_rate_limits.go:
```diff
@@ -61,7 +62,7 @@ func migrateRateLimits(appID string, configSourceData map[string]string, DryRun
 	}
 
-	_, err = config.Parse(migrated)
+	_, err = config.Parse(context.Background(), migrated)
```
Update internal.MigrateResources to pass ctx to UpdateConfigSourceFunc

louischan-oursky commented on pkg/lib/config/configsource/resources.go:
```diff
@@ -101,7 +101,8 @@ func (d AuthgearYAMLDescriptor) ViewResources(resources []resource.ResourceFile,
 		}
 
-		appConfig, err := config.Parse(bytes.([]byte))
+		// FIXME: the context should be passed from the caller
```
We should update the interface of Descriptor to make all methods take ctx as the first argument.

tung2744 replied:
Maybe I will only update `ViewResources` in this pr, because other methods seems not affected and require a lot of changes.

louischan-oursky commented on .vettedpositions:
```diff
@@ -317,3 +316,8 @@
+/cmd/portal/cmd/cmdinternal/migrate_resources_migrate_rate_limits.go:65:24: contextbackground
+/pkg/lib/config/configsource/resources.go:105:34: contextbackground
+/pkg/lib/config/configsource/resources.go:504:9: contextbackground
+/pkg/lib/config/configsource/resources.go:670:9: contextbackground
```
We should not allow `contextbackground` usage.

tung2744 replied:
Fixed, thanks!

tung2744 commented:
Rebased and updated. Changes start from [98326e0](https://github.com/authgear/authgear-server/pull/5017/commits/98326e04a447a20880af1658a01edd49a4947c2b)
